src/main/java/net/glowstone/block/state/GlowDispenser.java

```diff
@@ -68,17 +90,37 @@ public boolean dispense() {
 
     public int getDispenseSlot() {
         Inventory inv = getInventory();
-
-        int slot = -1;
-        int randomChance = 1;
-
-        for (int i = 0; i < inv.getSize(); i++) {
-            if (inv.getItem(i) != null && random.nextInt(randomChance++) == 0) {
-                slot = i;
+        
+        boolean isEmpty = true;
+        for (ItemStack stack : inv.getContents()) {
+            if (InventoryUtil.isEmpty(stack) == false) isEmpty = false;
```

Comment:
For efficiency, you should break your loop after finding a non-empty stack. Also, use `!InventoryUtil.isEmpty(stack)`.